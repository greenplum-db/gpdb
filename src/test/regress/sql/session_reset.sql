set log_min_messages to ERROR;
CREATE EXTENSION IF NOT EXISTS gp_inject_fault;
select gp_inject_fault ('dtm_broadcast_prepare', 'reset', 1);
SELECT gp_inject_fault ('dtm_broadcast_prepare', 'error', 1);
create table abc1 (id int, ctx text);
select gp_inject_fault ('dtm_broadcast_prepare', 'reset', 1);
select session_id AS session_not_in_stat_activity from (select paramvalue AS session_id from gp_toolkit.gp_param_settings() where paramname='gp_session_id' and paramsegment = 0) right_id where right_id.session_id not in (select sess_id::text from pg_stat_activity);
set log_min_messages to default;
