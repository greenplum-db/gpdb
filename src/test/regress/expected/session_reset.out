set log_min_messages to ERROR;
CREATE EXTENSION IF NOT EXISTS gp_inject_fault;
select gp_inject_fault ('dtm_broadcast_prepare', 'reset', 1);
NOTICE:  Success:
 gp_inject_fault 
-----------------
 t
(1 row)

SELECT gp_inject_fault ('dtm_broadcast_prepare', 'error', 1);
NOTICE:  Success:
 gp_inject_fault 
-----------------
 t
(1 row)

create table abc1 (id int, ctx text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'id' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
ERROR:  fault triggered, fault name:'dtm_broadcast_prepare' fault type:'error'
select gp_inject_fault ('dtm_broadcast_prepare', 'reset', 1);
NOTICE:  Success:
 gp_inject_fault 
-----------------
 t
(1 row)

select session_id AS session_not_in_stat_activity from (select paramvalue AS session_id from gp_toolkit.gp_param_settings() where paramname='gp_session_id' and paramsegment = 0) right_id where right_id.session_id not in (select sess_id::text from pg_stat_activity);
 session_not_in_stat_activity 
------------------------------
(0 rows)

set log_min_messages to default;
